// Copyright (C) 2020 THL A29 Limited, a Tencent company. All rights reserved.
//
// Licensed under the BSD 3-Clause License (the "License"); you may not use this
// file except in compliance with the License. You may obtain a copy of the
// License at
//
// https://opensource.org/licenses/BSD-3-Clause
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
// License for the specific language governing permissions and limitations under
// the License.

syntax = "proto3";

package flare.rpc;

option go_package = "tencent.com/protobuf/flare/rpc/protocol/protobuf/binlog_proto";

// Only responses are stored. We (`RpcChannel`) don't care about requests.
message SerializedClientPacket {
  int32 status = 1;
  bool streaming_rpc = 2;
  bool using_raw_bytes = 5;
  bytes body = 3;
  bytes attachment = 4;
};

// Only requests are stored.
message SerializedServerPacket {
  string method = 1;
  bool streaming_rpc = 2;
  bool using_raw_bytes = 5;
  bytes body = 3;
  bytes attachment = 4;
}
